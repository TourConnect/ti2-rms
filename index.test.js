const R = require('ramda')
const axios = require('axios');

const { typeDefs: productTypeDefs, query: productQuery } = require('./node_modules/ti2/controllers/graphql-schemas/product');
const { typeDefs: availTypeDefs, query: availQuery } = require('./node_modules/ti2/controllers/graphql-schemas/availability');
const { typeDefs: bookingTypeDefs, query: bookingQuery } = require('./node_modules/ti2/controllers/graphql-schemas/booking');
const { typeDefs: rateTypeDefs, query: rateQuery } = require('./node_modules/ti2/controllers/graphql-schemas/rate');
const { typeDefs: pickupTypeDefs, query: pickupQuery } = require('./node_modules/ti2/controllers/graphql-schemas/pickup-point');
const rmsBooking = require('./__apiReference__/booking.json');

const typeDefsAndQueries = {
  productTypeDefs,
  productQuery,
  availTypeDefs,
  availQuery,
  bookingTypeDefs,
  bookingQuery,
  rateTypeDefs,
  rateQuery,
  pickupTypeDefs,
  pickupQuery,
};

describe('auth tests', () => {
  let app;
  const token = {
    clientId: process.env.ti2_rms_clientId,
    clientPassword: process.env.ti2_rms_clientPassword,
  };
  beforeAll(async () => {
    const ti2 = await require('ti2')({
      plugins: {
        rms: require('./index'),
      },
      startServer: false,
    });
    app = ti2.plugins.find((plugin) => plugin.name === 'rms');
  });
  it('should validate an invalid token is ', async () => {
    const isValid = await app.validateToken({
      axios,
      token: {
        clientId: '123',
        clientPassword: 'bull-passw',
      }
    });
    expect(isValid).toBe(false);
  });
  it('should validate an valid token', async () => {
    const isValid = await app.validateToken({
      axios,
      token, 
    });
    expect(isValid).toBe(true);
  });
  it('should be able to get a list of affiliates', async () => {
    const { affiliates } = await app.getAffiliates({
      axios,
      token,
    });
    expect(Array.isArray(affiliates)).toBe(true);
  }, 30e3);
  it('should be able to get a list or booking fields', async () => {
    let testProduct = {
      productName: 'Test property 1',
    };
    const retVal = await app.getCreateBookingFields({
      axios,
      token,
      typeDefsAndQueries,
    });
    expect(Array.isArray(retVal.productFields)).toBeTruthy();
    expect(retVal.productFields.length).toBe(8);
    expect(retVal.productFields.map(p => p.id)).toMatchSnapshot();
    expect(retVal.productFields[0].options.find(o => o.label === testProduct.productName)).toBeTruthy();
  }, 30e3);
  it('should be able to translate a booking', async () => {
    const booking = await app._translateBooking({
      axios,
      rmsBooking,
      token,
      bookingTypeDefs,
      bookingQuery,
    })
    expect(booking).toMatchSnapshot();
  });
});
