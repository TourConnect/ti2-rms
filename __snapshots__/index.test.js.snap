// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`auth tests should be able to get a list or booking fields 1`] = `
[
  "propertyId",
  "startDate",
  "endDate",
  "rateId",
  "categoryId",
  "adults",
  "children",
  "infants",
]
`;

exports[`auth tests should be able to translate a booking 1`] = `
{
  "adults": 1,
  "agent": null,
  "allDay": false,
  "bookingDate": "2023-06-24 05:04:20",
  "bookingId": "215718",
  "cancelPolicy": null,
  "cancellable": true,
  "categoryId": 1,
  "children": 0,
  "desk": null,
  "editable": true,
  "end": "2023-06-24 23:59:59",
  "endDate": "2023-06-24 23:59:59",
  "holder": {
    "emailAddress": "",
    "fullName": "Vic Mann",
    "name": "Vic",
    "phoneNumber": "",
    "surname": "Mann",
  },
  "id": "215718",
  "infants": 0,
  "notes": "",
  "optionId": null,
  "optionName": null,
  "orderId": "215718",
  "pickupHotel": null,
  "pickupHotelRoom": null,
  "pickupPoint": null,
  "pickupPointId": null,
  "pickupRequested": null,
  "price": {
    "currency": null,
    "currencyPrecision": null,
    "net": null,
    "original": null,
    "retail": null,
  },
  "privateUrl": null,
  "productId": null,
  "productName": null,
  "propertyId": 1,
  "publicUrl": null,
  "rateId": 1159,
  "resellerReference": "",
  "start": "2023-06-23 00:00:00",
  "startDate": "2023-06-23 00:00:00",
  "status": "Unconfirmed",
  "supplierBookingId": "215718",
  "unitItems": [],
}
`;
